src/Spice86.Shared/Diagnostics/PerformanceMeasurer.cs

```diff
@@ -31,21 +31,36 @@ public class PerformanceMeasurer : IPerformanceMeasurer {
     /// <inheritdoc />
     public void UpdateValue(long newMeasure) {
         long newTimeInMilliseconds = GetCurrentTime();
-        if (_firstMeasureTimeInTicks == 0) {
-            _firstMeasureTimeInTicks = newTimeInMilliseconds;
-        } else if((TimeSpan.FromTicks(newTimeInMilliseconds) - TimeSpan.FromTicks(_firstMeasureTimeInTicks)).TotalSeconds >= WindowSizeInSeconds) {
-            _firstMeasureTimeInTicks = newTimeInMilliseconds;
-            _sampledMetricsCount = 0;
-            AverageValuePerSecond = 0;
+        if (IsFirstMeasurement()) {
+            _firstMeasureTimeInMilliseconds = newTimeInMilliseconds;
+        } else if (IsLastMeasurementExpired(newTimeInMilliseconds)) {
+            ResetMetrics(newTimeInMilliseconds);
         }
+
         long millisecondsDelta = newTimeInMilliseconds - _lastTimeInMilliseconds;
+        if (millisecondsDelta == 0) {
+            return;
+        }
         _lastTimeInMilliseconds = newTimeInMilliseconds;
         long valueDelta = newMeasure - _measure;
         _measure = newMeasure;
         ValuePerMillisecond = valueDelta / Math.Max(millisecondsDelta, 1);
```

Comment:
Since you added a check for 0 above, you can get rid of the Max()
```suggestion
        ValuePerMillisecond = valueDelta / millisecondsDelta;
```